Tests/xSharePoint/xSharePoint.xSPAccessServiceApp.Tests.ps1

```diff
@@ -0,0 +1,99 @@
+Import-Module (Join-Path $RepoRoot "Modules\xSharePoint\DSCResources\$ModuleName\$ModuleName.psm1")
+
+Describe "xSPAccessServiceApp" {
+    InModuleScope $ModuleName {
+        $testParams = @{
+            Name = "Test Access Services App"
+            DatabaseName = "SP_Access"
+            ApplicationPool = "Test App Pool"
+        }
+        Import-Module (Join-Path ((Resolve-Path $PSScriptRoot\..\..).Path) "Modules\xSharePoint")
+
+        
+        Mock Invoke-xSharePointCommand { 
+            return Invoke-Command -ScriptBlock $ScriptBlock -ArgumentList $Arguments -NoNewScope
+        }
+        
+        Import-Module $Global:CurrentSharePointStubModule -WarningAction SilentlyContinue
+
+        Context "When no service applications exist in the current farm" {
+
+            Mock Get-SPServiceApplication { return $null }
+            Mock New-SPAccessServicesApplication { }
+
+            It "returns null from the Get method" {
+                Get-TargetResource @testParams | Should BeNullOrEmpty
+                Assert-MockCalled Get-SPServiceApplication -ParameterFilter { $Name -eq $testParams.Name } 
+            }
+
+            It "returns false when the Test method is called" {
+                Test-TargetResource @testParams | Should Be $false
+            }
+
+            It "creates a new service application in the set method" {
+                Set-TargetResource @testParams
+                Assert-MockCalled New-SPAccessServicesApplication 
+            }
+        }
+
+        Context "When service applications exist in the current farm but the specific Access Services app does not" {
+
+            Mock Get-SPServiceApplication { return @(@{
+                TypeName = "Some other service app type"
+            }) }
+
+            It "returns null from the Get method" {
+                Get-TargetResource @testParams | Should BeNullOrEmpty
+                Assert-MockCalled Get-SPServiceApplication -ParameterFilter { $Name -eq $testParams.Name } 
+            }
+
+        }
+
+        Context "When a service application exists and is configured correctly" {
+            Mock Get-SPServiceApplication { 
+                return @(@{
+                    TypeName = "Access Services Application"
+                    DisplayName = $testParams.Name
+                    DatabaseName = $testParams.DatebaseName
+                    ApplicationPool = @{ Name = $testParams.ApplicationPool }
+                })
+            }
+
+            It "returns values from the get method" {
+                Get-TargetResource @testParams | Should Not BeNullOrEmpty
+                Assert-MockCalled Get-SPServiceApplication -ParameterFilter { $Name -eq $testParams.Name } 
+            }
+
+            It "returns true when the Test method is called" {
+                Test-TargetResource @testParams | Should Be $true
+            }
+        }
+
+        Context "When a service application exists and is not configured correctly" {
```

Comment:
There should be a test here of the set method to make sure it updates the app pool